### PR TITLE
Add support for episode-level podcast:value tags

### PR DESCRIPTION
**Pull request by dnlggr**
I've been trying to get support for per-episode payment splits working in Breez as per the [Podcastindex.org specs](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md#value-element) for the `"podcast:value"` element.

Currently, Breez only parses channel-level value definitions which allows for definitions of payment schemes for an entire podcast. The [spec](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md#value-element) also allows for item-level value definitions which allows for different payment splits for individual episodes (effectively overriding the global channel-level value definition for the respective episode).

In order to do that, we need access to episode metadata containing possible `"podcast:value"` definitions that might be defined on an episode level. I've followed [the implementation](https://github.com/breez/anytime_podcast_player/commit/e35d10c9fdbb6e5c9b99a208c25d72a716e2b604) of the podcast-level metadata hook and added a second one for loading episode-level metadata.

This pull request goes hand-in-hand with a second one in breezmobile that will make use of the episode level metadata in Breez: https://github.com/breez/breezmobile/pull/591

A possible concern with this implementation is that we might be loading large amounts of metadata for podcasts that have many episodes. I've been thinking about patching the underlying podcast-search library to include metadata when it downloads the episode. Not sure if that would be overkill for now, though. Let me know what you think!

This is my first time writing Dart—don't hesitate to be extra strict with my code. 😄

### CODE REVIEW

**Comment by dnlggr** [comment]
Ok, so after some thinking I want to ask you for your opinion on the best way to implement this @roeierez.

I definitely think we should make the metadata loading more efficient. To query the episode metadata at the PodcastIndex.org backend, we need access to the [PodcastIndex.org episode IDs](https://podcastindex-org.github.io/docs-api/#get-/episodes/byid) (not the GUIDs but the PodcastIndex.org specific IDs).

After some digging around, it seems like we don't have access to the PodcastIndex.org episode IDs at the point we want to play them. As far as I can tell, this is because the feed gets parsed by the [podcast-search](https://github.com/amugofjava/podcast_search) library which directly uses the feed URL provided by the PodcastIndex.org search to load a podcast's RSS feed.
Do you know if we somewhere already have access to those PodcastIndex.org episode IDs? I couldn't find a place where we do, but I'm not that confident with the code base so maybe just I'm missing it.
If we don't already have them we would have to query the [Load Episodes By Feed URL](https://podcastindex-org.github.io/docs-api/#get-/episodes/byfeedurl) endpoint — which would be equivalent to what this PR currently does. So that wouldn't be a solution to the problem.

Another option of course would be to drop down to the _podcast-search_ library itself and just parse the metadata directly from the RSS feed when we load it for the first time. This wouldn't give us access to the PodcastIndex.org IDs but would instead give us the metadata directly — without needing to query the PodcastIndex.org backend later. However, it would mean modifying and possibly forking another dependency.

Best case would be if we could get ahold of the PodcastIndex.org episode IDs somehow without loading all of the episode metadata at once like we currently do in this PR.

What do you think? Am I missing something?

**Comment by dnlggr** [comment]
Update: I have the commit for `podcast-search` ready here: https://github.com/dnlggr/podcast_search/commit/11c644fb16a40d4205b15e95ae39a4b69a68808f.

Once https://github.com/sudame/dart-rss/pull/17 gets merged, I can open that PR to integrate `podcast:value` into episode and podcast models. Then we can use it in our anytime fork here and finally integrate it into breezmobile...

Let's hope https://github.com/sudame/dart-rss/pull/17 gets merged soon. 🙏

**Comment by dnlggr** [comment]
Alright so I updated this PR based on the changes in https://github.com/breez/podcast_search/pull/2.

Once that is merged, we can update the `pubspec.yaml` and then this PR should be ready for review. Let me know what you think!

**Comment by dnlggr** [comment]
Alright I think this should be ready for review now @roeierez. 👍

**Comment by roeierez** [comment]
@dnlggr  I think currently there could be cases where episode value is only retrieved when using podcast index API and the value tag doesn't exist in the feed. I think it is reasonable to start with this approach given it simplifies a lot and see how it goes. Do you have some examples you can share for existing live episodes that has the value tags defined in their feed? Just wanted to make sure people use the feed for that and not only podcast index API.

**Comment by dnlggr** [comment]
> I think it is reasonable to start with this approach given it simplifies a lot and see how it goes.

Agreed! We can always build on top of what we have at a later point if needed. Supporting the value tag from the feed should be a good start.

> Do you have some examples you can share for existing live episodes that has the value tags defined in their feed?

Yeah, here: https://closing-the-loop.github.io/feed.xml

> Just wanted to make sure people use the feed for that and not only podcast index API.

To be honest, I don't even know if they support per-episode payment definitions there. I'll try to find out and report back — but as I said, I think we can always build on top of what we have now later (if needed).

Let me know what you think! Just wanted to bring this up for a brief discussion as I didn't really know if it was relevant or not.

File: lib/services/podcast/mobile_podcast_service.dart [hunk 1]
```diff
@@ -31,7 +32,7 @@ class MobilePodcastService extends PodcastService {
     @required PodcastApi api,
     @required Repository repository,
     @required SettingsService settingsService,
-    this.loadMetadata,
+    this.loadMetadata, // Podcast metadata
```
[the remark this answers is not in the document]
Comment (roeierez) [reply lib/services/podcast/mobile_podcast_service.dart:35]: @dnlggr excellent communication on this an its dependant PRs. Definitely helps pushing them a lot faster.

**Review by roeierez — APPROVED** [review]

**Pull request merged**